src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,332 @@
+      </Modal>
+      {DisplayComponent}
+    </>
+  )
+}
+
+export default ApplicationView
+
+const createTextString = (listItems: ListItem[]) => {
+  // This doesn't really get used except to identify whether the element has been entered or not
+  // TO-DO
+  return 'Temp text value'
+}
+
+const resetCurrentResponses = (inputFields: TemplateElement[]) =>
+  new Array(inputFields.length).fill({ value: { text: undefined } })
+
+const buildListItem = (item: ListItem, inputFields: TemplateElement[]) =>
+  item.map((field, index) => ({
+    code: inputFields[index].code,
+    id: field.id,
+    title: inputFields[index].title,
+    value: field.value,
+  }))
+
+const anyInvalidItems = (currentInput: InputResponseField[]) =>
+  currentInput.some((response) => response.isValid === false)
+
+const anyIncompleteItems = (currentInput: InputResponseField[], inputFields: TemplateElement[]) =>
+  currentInput.some(
+    (response, index) => inputFields[index]?.isRequired !== false && !response.value.text
+  )
+
+const anyErrorItems = (currentInput: InputResponseField[], inputFields: TemplateElement[]) =>
+  anyInvalidItems(currentInput) || anyIncompleteItems(currentInput, inputFields)
+
+const substituteValues = (parameterisedString: string, item: any) => {
+  const getValueFromCode = (_: string, $: string, code: string) =>
+    item.find((field: any) => field.code === code).value.text || ''
+  return parameterisedString.replace(/(\${)(.*?)(})/gm, getValueFromCode)
+}
+
+// Separate components, so can be shared with SummaryView
+export interface ListLayoutProps {
+  listItems: ListItem[]
+  displayFormat: { header?: string; meta?: string; description: string }
+  Markdown: any
+  fieldTitles?: string[]
+  editItem?: Function
+  deleteItem?: Function
+  isEditable?: boolean
+}
+
+export const ListCardLayout: React.FC<ListLayoutProps> = ({
+  listItems,
+  displayFormat,
+  editItem = () => {},
+  deleteItem = () => {},
+  Markdown,
+  isEditable = true,
+}) => {
+  const { header, meta, description } = displayFormat
+  return (
+    <>
+      {listItems.map((item, index) => (
+        <Card key={`list-item-${index}`}>
+          <Card.Content>
+            {isEditable && (
+              <Label floating onClick={() => deleteItem(index)}>
+                <Icon name="delete" />
+              </Label>
+            )}
+            {header && (
+              <Card.Header onClick={() => editItem(index)}>
+                <Markdown text={substituteValues(header, item)} semanticComponent="noParagraph" />
+              </Card.Header>
+            )}
+            {meta && (
+              <Card.Meta onClick={() => editItem(index)}>
+                <Markdown text={substituteValues(meta, item)} semanticComponent="noParagraph" />
+              </Card.Meta>
+            )}
+            {description && (
+              <Card.Description onClick={() => editItem(index)}>
+                <Markdown
+                  text={substituteValues(description, item)}
```

Comment:
I wonder if we really need this, or if we can just have description with `Title` and `Value`. I can see where you were going with dynamic card formatting, but not sure if it's really needed

Reply:
I think we need some kind of substitution system, and this seemed the simplest to me. But we can discuss options. (But if a card is only ever going to show "Title: Value", then there's not really much point displaying cards, you may as well always show Table).

Reply:
Have added a default/fallback -- if "displayFormat" is not provided,  the card will just show "Title: Value" as you describe.

As you'll see, it looks _uuuugly_, so it's nice to have the fancy option too, if configurer can be bothered. :) 

86212533